app/assets/scripts/components/upload-form.js

```diff
@@ -125,59 +124,67 @@ var UploadForm = React.createClass({
         v.errors.forEach((e) => {
           failures.push(`Record ${line}: ${e.stack}`);
         });
+        if (line === 0) {
+          // Add static information to metadata on first line
+          metadata.location = record.location;
+          metadata.city = record.city;
+          metadata.country = record.country;
+          metadata.dates = {};
+          metadata.values = {};
+        }
+        // Add array information to metadata
+        metadata.dates[record.date.local] = true;
+        metadata.values[record.value] = true;
+
+        line++;
       })
       .on('end', () => {
-        this.logOutput(failures);
+        metadata.measurements = line;
+        this.logOutput(failures, metadata);
       });
   },
 
-  storeCsv: function (event) {
-    this.csvData = event.target.files[0];
+  getFile: function (event) {
+    // Store file reference
+    this.csvFile = event.target.files[0];
+    this.setState({formFile: this.csvFile.name});
   },
 
   render: function () {
     const errors = this.state.errors;
     const showModal = this.state.showModal;
     return (
       <section className='fold' id='uploader'>
-        {showModal
-          ? <FailureModal errors={errors} /> : ''}
-        {showModal
-          ? <SuccessModal visible={showModal} errors={errors} />
-          : ''}
-        <div className='inner'>
-          <header className='fold__header'>
-            <h1 className='fold__title'>OpenAQ Uploader</h1>
-            <div className='fold__introduction prose prose--responsive'>
-              <p>Use this form to upload CSV data to OpenAQ</p>
-            </div>
-          </header>
-          <div className='fold__body'>
-            The CSV parser is able to recognize the following attribute names and characteristics:
-            (An example schema should be included, potentially as a hidden-by-default accordian list)
-          </div>
-
-          <fieldset className='form__fieldset'>
+        <div className='exhibit'>
+          <div className="exhibit__content">
+            {showModal ? <FailureModal errors={errors} /> : ''}
+            {showModal ? <SuccessModal visible={showModal} metadata={this.state.metadata} errors={errors} csvFile={this.csvFile} /> : ''}
```

Comment:
You should be using the modals from the design system.
The design for this page doesn't actually feature modals, but just for reference here's an implementation: https://github.com/openaq/openaq-design-system/blob/master/sandbox/assets/scripts/main.js#L32-L55